flink-connectors/flink-connector-gcp-pubsub/src/main/java/org/apache/flink/streaming/connectors/gcp/pubsub/source/enumerator/PubSubEnumeratorCheckpointSerializer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.gcp.pubsub.source.enumerator;
+
+import org.apache.flink.core.io.SimpleVersionedSerializer;
+
+import java.io.IOException;
+
+/**
+ * A stub to serialize the contents of a {@link PubSubEnumeratorCheckpoint}. Because no data is
+ * stored in such a checkpoint, no proper serialization is necessary.
+ */
+public class PubSubEnumeratorCheckpointSerializer
+        implements SimpleVersionedSerializer<PubSubEnumeratorCheckpoint> {
+
+    private static final int CURRENT_VERSION = 0;
+
+    @Override
+    public int getVersion() {
+        return CURRENT_VERSION;
+    }
+
+    @Override
+    public byte[] serialize(PubSubEnumeratorCheckpoint enumeratorCheckpoint) throws IOException {
+        return new byte[0];
+    }
+
+    @Override
+    public PubSubEnumeratorCheckpoint deserialize(int version, byte[] serialized)
```

Comment:
To allow state evolution in the future it is important to check the version https://github.com/apache/flink/blob/bebf3b5a105dd4bc21882116570c6d71299269a6/flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/enumerator/KafkaSourceEnumStateSerializer.java#L52.

A simple check is enough that the version is equal to the one we are expecting.